This patch updates enum-as-inner to 0.6 and removes the duplicate

It is based on https://github.com/hickory-dns/hickory-dns/pull/1948

Index: trust-dns-proto/Cargo.toml
===================================================================
--- trust-dns-proto.orig/Cargo.toml
+++ trust-dns-proto/Cargo.toml
@@ -65,7 +65,7 @@ version = "1"
 version = "2.2.0"
 
 [dependencies.enum-as-inner]
-version = "0.5"
+version = "0.6"
 
 [dependencies.futures-channel]
 version = "0.3.5"
Index: trust-dns-proto/src/rr/record_data.rs
===================================================================
--- trust-dns-proto.orig/src/rr/record_data.rs
+++ trust-dns-proto/src/rr/record_data.rs
@@ -990,10 +990,6 @@ impl RData {
         }
     }
 
-    /// Returns true if
-    pub fn is_soa(&self) -> bool {
-        matches!(self, RData::SOA(..))
-    }
 }
 
 impl fmt::Display for RData {
